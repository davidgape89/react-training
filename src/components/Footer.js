import React from 'react';

import './Footer.scss';

const Footer = (props) => (
  <div className="footer">
    netflixroulette
  </div>
);

export {Footer};