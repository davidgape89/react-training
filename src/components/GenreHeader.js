import React from 'react';
import PropTypes from 'prop-types';

const GenreHeader = ({genre}) => (
  <div className="genre-header">
    <div className="genre-header__genre">
      Films by {genre}
    </div>
  </div>
)

GenreHeader.propTypes = {
  genre: PropTypes.string
};

export default GenreHeader;