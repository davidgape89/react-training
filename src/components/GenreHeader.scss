@import '../styles/variables';

.genre-header {
  align-items: center;
  background-color: $light-grey;
  display: flex;
  height: 50px;
  padding: 0 50px;
  width: $main-layout-width;
}