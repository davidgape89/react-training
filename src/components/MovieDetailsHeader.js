import React from 'react';
import PropTypes from 'prop-types';

import './MovieDetailsHeader.scss';

const MovieDetailsHeader = ({
  title,
  genres,
  releaseDate,
  posterUrl,
  runtime,
  voteAverage,
  overview
}) => {
  const genre = genres.join(' & ');
  const year = releaseDate.split('-')[0];
  return (
    <div className="movie-details-header">
      <div className="movie-details-header__header">
        <div>netflixroulette</div>
        <button className="button button--white">Search</button>
      </div>
      <div className="movie-details-header__details">
        <div className="movie-details-header__image">
          <img src={posterUrl} />
        </div>
        <div className="movie-details-header__info">
          <div className="movie-details-header__first-row">
            <div className="movie-details-header__title">
              {title}
            </div>
            <div className="movie-details-header__vote-average">
              {voteAverage}
            </div>
          </div>
          <div>{genre}</div>
          <div className="movie-details-header__third-row">
            <span>{year}</span>
            <span>{runtime} min</span>
          </div>
          
          <div>{overview}</div>
        </div>
      </div>
    </div>
  );
}

MovieDetailsHeader.propTypes = {
  title: PropTypes.string,
  genres: PropTypes.array,
  releaseDate: PropTypes.string,
  posterUrl: PropTypes.string,
  runtime: PropTypes.number,
  voteAverage: PropTypes.number,
  overview: PropTypes.string
};

export {MovieDetailsHeader};