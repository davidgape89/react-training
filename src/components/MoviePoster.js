import React from 'react';
import PropTypes from 'prop-types';

const MoviePoster = ({
  id,
  title, 
  genres,
  releaseDate,
  posterUrl,
  handleMovieSelected
}) => {
  const year = releaseDate.split('-')[0];
  const genre = genres.join(' & ');
  return (
    <div className="movie-poster" 
         onClick={() => handleMovieSelected(id)}>
      <img src={posterUrl} />
      <div>
        <div className="movie-poster__info">
          <div className="title">{title}</div>
          <div className="year">{year}</div>
        </div>
        <div className="genre">{genre}</div>
      </div>
    </div>
  );
}

MoviePoster.propTypes = {
  title: PropTypes.string,
  genres: PropTypes.array,
  releaseDate: PropTypes.string,
  posterUrl: PropTypes.string
};

export default MoviePoster;