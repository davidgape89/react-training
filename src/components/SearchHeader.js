import React from 'react';
import {connect} from 'react-redux';
import {startMoviesRequest} from '../actions/movies';
import SearchBar from './SearchBar';
import SearchByToggle from './SearchByToggle';

export class SearchHeader extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      query: '',
      searchBy: 'title'
    }
  }

  onQueryChange = (event) => {
    this.setState({query: event.target.value});
  }

  onKeyPress = (event) => {
    if(event.key === "Enter") {
      this.search();
    }
  }

  changeSearchBy = (criteria) => {
    this.setState({searchBy: criteria});
  }

  search = () => {
    const {query, searchBy} = this.state;
    this.props.handleSearch(query, searchBy);
  }

  render() {
    return (
      <div className="search-header">
        <div className="search-header__title">
          netflixroulette
        </div>
        <div className="search-header__search-bar">
          <label>FIND YOUR MOVIE</label>
          <SearchBar query={this.state.query} 
                      placeholder={'Some movie title or genre'}
                      onChange={this.onQueryChange}
                      onKeyPress={this.onKeyPress} />
        </div>
        <div className="search-header__bottom-bar">
          <SearchByToggle value={this.state.searchBy}
                          onChange={this.changeSearchBy}/>
          <button className="button button--red" 
                  onClick={this.search}>SEARCH</button>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = {
  handleSearch: startMoviesRequest
};

export default connect(null, mapDispatchToProps)(SearchHeader);