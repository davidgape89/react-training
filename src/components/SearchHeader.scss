@import '../styles/variables';

$opacity-rgba: rgba(0, 0, 0, 0.7);

.search-header {
  background: linear-gradient($opacity-rgba, $opacity-rgba), 
              url('/assets/netflix-background.jpg');
  display: flex;
  justify-content: center;
}

.search-header__content {
  min-width: 900px;
  padding: $big-size;
}

.search-header__title {
  color: $light-red;
  font-weight: 600;
  margin-bottom: 50px;
}

.search-header__search-bar {
  display: flex;
  flex-direction: column;

  label {
    color: $white;
    font-weight: 600;
    margin-bottom: $medium-size;
  }
}

.search-header__bottom-bar {
  display: flex;
  justify-content: space-between;
  margin-top: $medium-size;
}