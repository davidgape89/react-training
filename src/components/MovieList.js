import React from 'react';
import MoviePoster from './MoviePoster';

export default ({movies, handleMovieSelected}) => (
  <div className="movie-list">
    {movies.map((movie) => (
      <MoviePoster key={movie.id}
                  id={movie.id}
                  title={movie.title}
                  genres={movie.genres}
                  posterUrl={movie.poster_path}
                  releaseDate={movie.release_date}
                  handleMovieSelected={handleMovieSelected} />
    ))}
  </div>
);