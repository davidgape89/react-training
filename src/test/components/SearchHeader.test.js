import React from 'react';
import {shallow} from 'enzyme';

import {SearchHeader} from '../../components/SearchHeader';

describe('SearchHeader -', () => {
  let wrapper, reactRouterMock, location, match, history;
  
  beforeEach(() => {
    reactRouterMock = {
      location: {
        search: '?something=else'
      }
    };
    location = {
      search: null
    };
    match = {
      params: {}
    }
    history = {
      push: jest.fn()
    }
    wrapper = shallow(<SearchHeader 
      location={location}
      match={match}
      history={history}/>
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('changes the query correctly', () => {
    const mockEvent = {
      target: {
        value: 'newquery'
      }
    };
    wrapper.find('SearchBar').prop('onChange')(mockEvent);

    expect(wrapper.state('query')).toBe('newquery');
  });

  it('calls search function when enter is pressed', () => {
    const mockEvent = {
      key: 'Enter'
    };
    wrapper.find('SearchBar').prop('onKeyPress')(mockEvent);

    expect(history.push).toHaveBeenCalled();
  });

  it('does not call search function when other keys are pressed', () => {
    const mockEvent = {
      key: 'j'
    };
    wrapper.find('SearchBar').prop('onKeyPress')(mockEvent);

    expect(history.push).not.toHaveBeenCalled();
  });

  it('changes search by correctly', () => {
    wrapper.find('SearchByToggle').prop('onChange')('genre');

    expect(wrapper.state('searchBy')).toBe('genre');
  });
});