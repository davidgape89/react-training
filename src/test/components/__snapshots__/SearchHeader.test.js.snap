// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader - renders correctly 1`] = `
<div
  className="search-header"
>
  <div
    className="search-header__title"
  >
    netflixroulette
  </div>
  <div
    className="search-header__search-bar"
  >
    <label>
      FIND YOUR MOVIE
    </label>
    <SearchBar
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Some movie title or genre"
      query=""
    />
  </div>
  <div
    className="search-header__bottom-bar"
  >
    <SearchByToggle
      onChange={[Function]}
      value="title"
    />
    <Link
      to="/search/?searchBy=title"
    >
      <button
        className="button button--red"
      >
        SEARCH
      </button>
    </Link>
  </div>
</div>
`;
